server/src/main/java/org/elasticsearch/index/engine/SegmentsStats.java

```diff
@@ -29,7 +32,7 @@ public class SegmentsStats implements Writeable, ToXContentFragment {
     private long versionMapMemoryInBytes;
     private long maxUnsafeAutoIdTimestamp = Long.MIN_VALUE;
     private long bitsetMemoryInBytes;
-    private ImmutableOpenMap<String, FileStats> files = ImmutableOpenMap.of();
+    private Map<String, FileStats> files = new HashMap<>();
```

Comment:
Related to the other constructor comment. Make this `final` but assign in the constructors so we can pre-size it when reading from `StreamInput`?